smoke-tests-otel-starter/src/test/java/io/opentelemetry/smoketest/OtelSpringStarterSmokeTest.java

```diff
@@ -56,28 +73,80 @@ class OtelSpringStarterSmokeTest {
 
   @Autowired private TestRestTemplate testRestTemplate;
 
-  @Autowired private ConfigProperties configProperties;
+  @Autowired private Environment environment;
+  @Autowired private PropagationProperties propagationProperties;
+  @Autowired private OtelResourceProperties otelResourceProperties;
+  @Autowired private OtlpExporterProperties otlpExporterProperties;
 
   @Configuration(proxyBeanMethods = false)
   static class TestConfiguration {
+
+    @Bean
+    ConfigurableMetricExporterProvider otlpMetricExporterProvider() {
+      return new ConfigurableMetricExporterProvider() {
+        @Override
+        public MetricExporter createExporter(ConfigProperties configProperties) {
+          return METRIC_EXPORTER;
+        }
+
+        @Override
+        public String getName() {
+          return "memory";
+        }
+      };
+    }
+
     @Bean
-    public MetricExporter metricExporter() {
-      return METRIC_EXPORTER;
+    ConfigurableSpanExporterProvider otlpSpanExporterProvider() {
+      return new ConfigurableSpanExporterProvider() {
+        @Override
+        public SpanExporter createExporter(ConfigProperties configProperties) {
+          return SPAN_EXPORTER;
+        }
+
+        @Override
+        public String getName() {
+          return "memory";
+        }
+      };
     }
 
     @Bean
-    public SpanExporter spanExporter() {
-      return SPAN_EXPORTER;
+    ConfigurableLogRecordExporterProvider otlpLogRecordExporterProvider() {
+      return new ConfigurableLogRecordExporterProvider() {
+        @Override
+        public LogRecordExporter createExporter(ConfigProperties configProperties) {
+          return LOG_RECORD_EXPORTER;
+        }
+
+        @Override
+        public String getName() {
+          return "memory";
+        }
+      };
     }
 
     @Bean
-    public LogRecordExporter logRecordExporter() {
-      return LOG_RECORD_EXPORTER;
+    AutoConfigurationCustomizerProvider propagatorCustomizer() {
```

Comment:
Where is this bean injected?

Reply:
OpenTelemetrySdkComponentLoader loads it from the spring `Environment`

Reply:
I don't see the use of a Spring `Environment` object in `OpenTelemetrySdkComponentLoader` and I don't use the use of `AutoConfigurationCustomizerProvider`  bean. Could you please elaborate?

Reply:
- SDK loads `AutoConfigurationCustomizerProvider` [here](https://github.com/open-telemetry/opentelemetry-java/blob/4519a7e90243e5b75b3a46a14c872de88b95a9a1/sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdkBuilder.java#L405-L408)
- component loader loads the bean [here](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/4727d87dbae2837f2494bf317e69265edc1a312b/instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/OpenTelemetryAutoConfiguration.java#L138) (sorry, it's not `Environment` - I mixed that up)

Reply:
done